core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/S3BatchSource.java

```diff
@@ -72,25 +84,47 @@ private static String updateFileSystemProperties(@Nullable String fileSystemProp
   public static class S3BatchConfig extends FileBatchConfig {
     @Description(ACCESS_ID_DESCRIPTION)
     @Macro
+    @Nullable
     private final String accessID;
 
     @Description(ACCESS_KEY_DESCRIPTION)
     @Macro
+    @Nullable
     private final String accessKey;
 
-    public S3BatchConfig(String referenceName, String accessID, String accessKey, String path) {
-      this(referenceName, accessID, accessKey, path, null, null, null, null, null, false, false);
+    @Description(AUTHENTICATION_METHOD)
```

Comment:
Why doesn't this plugin let me enter the IAMRole name like the other S3 plugins?